### PR TITLE
Stylus linting

### PR DESCRIPTION
**Pull request by nicksellen**
## What does this PR do?

For some reason I thought to try adding a stylus linter again. I should be doing something else though.

It sort of worked!

I would have liked an eslint-based one, but that doesn't seem to exist. This uses a postcss approach... and this experimental stylus parser [postcss-styl](https://www.npmjs.com/package/postcss-styl).

It has:
- `yarn lint:style`
- `yarn fix:style`
- webpack dev linting with the usual nice overlay thing

## Links to related issues

Also see:
#1154 
#1156 

## Checklist

- [ ] added a test, or explain why one is not needed/possible...
- [ ] no unrelated changes
- [ ] asked someone for a code review
- [ ] joined #karrot-dev channel at https://slackin.yunity.org
- [ ] added an entry to CHANGELOG.md (description, pull request link, username(s))

### CODE REVIEW

File: src/base/pages/Root.vue [hunk 1]
```diff
@@ -34,11 +34,12 @@ export default {
 
 <style lang="stylus" scoped>
 .root-splash-second
+  opacity 1;
```
Comment (nicksellen) [inline src/base/pages/Root.vue]: it adds a `;` when it reorders them :-1:

Comment (nicksellen) [reply src/base/pages/Root.vue]: hmmm... I hoped the "no-trailing-semicolons" rule had fixed it, but that just means no semicolons at the end of a block. The problem is the parser is stylus, but the rules are not designed for stylus....

File: src/pickups/components/PickupItem.vue [hunk 2]
```diff
@@ -116,25 +116,23 @@ export default {
...
+
   &.isEmpty:not(.isDisabled)
-    background repeating-linear-gradient(
-      135deg,
-      white,
-      white 15px,
-      $lightRed 15px,
-      $lightRed 30px
-    )
+    background repeating-linear-gradient(     135deg,     white,     white 15px,     $lightRed 15px,     $lightRed 30px   )
```
Comment (nicksellen) [inline src/pickups/components/PickupItem.vue]: Oo-er that's ugly

**Comment by nicksellen** [comment]
Whoops I wrote two custom plugins to get it to look right. Should definitely have not been working on that.

File: src/pickups/components/PickupItem.vue (hunk 2 above)
Comment (tiltec) [reply src/pickups/components/PickupItem.vue]: Why is there so much whitespace?

File: .circleci/config.yml [hunk 3]
```diff
@@ -70,7 +70,7 @@ jobs:
           keys:
           - v1-dependencies-{{ checksum "package.json" }}
           - v1-dependencies-
-      - run: yarn
+      - run: yarn --ignore-engines
```
Comment (tiltec) [inline .circleci/config.yml]: Why would the engine check fail?

Comment (nicksellen) [reply .circleci/config.yml]: The postcss-styl library has an oddly specific engines configuration, probably not intended... but we had something like 10.15...

https://github.com/ota-meshi/postcss-styl/blob/647780fb3d40e497c53d518bf60145c5ed69ad2a/package.json#L37

**Comment by nicksellen** [comment]
Yes, ready to merge! Except for the yarn.lock conflict...

**Pull request merged**